specs/eventing/overview.md

```diff
@@ -1,97 +1,88 @@
-implementation as part of the generic Channel (using a _channelTemplate_).
+## Messaging
 
-## Trigger
+### Channel
 
-**Trigger** describes a registration of interest on a filter set of events
-delivered to a _Broker_ which should be delivered to an _Addressable_. Events
-selected by a _Trigger_ are buffered independently from other _Triggers_, even
-if they deliver to the same _Addressable_.
+**Channel** provides an abstract interface which may be fulfiled by several
+concrete implementations of a backing asynchronous fan-out queue. The common
+abstraction provided by channel allows both the composition of higher-level
+constructs for chained or parallel processing of events, and the replacement of
+particular messaging technologies (for example, allowing a development
+environment to use a lower-reliability channel compared with the production
+environment).
 
-For more details, see [Kind: Trigger](spec.md#kind-trigger).
+### Subscription
 
-## Broker
+**Subscription** defines a delivery destination for all events sent to a
+**Channel**. Events sent to a channel are delivered to _each_ subscription
+_independently_ -- a subscription maintains its own list of undelivered events
+and will manage retry indpendently of any other subscriptions to the same
+channel. Like **Trigger**, subscriptions use the **Addressable** interface
+contract to support event delivery to many different destination types.
 
-**Broker** provides an eventing mesh. This allows producers to deliver events to
-a single endpoint and not need to worry about the routing details for individual
-consumers.
+## Interface Contracts
 
-For more details, see [Kind: Broker](spec.md#kind-broker).
+In addition to the concrete types described above in the `messaging.knative.dev`
+and `eventing.knative.dev` API groups, Knative Eventing supports referencing
+objects in other API groups as destinations for event delivery. This is done by
+defining partial schemas which the other resources must support. The following
+interface contracts define a set of expected resource fields on an referenced
+resource.
 
-## Subscription
+### Addressable
 
-**Subscriptions** describe a flow of events from one _Channel_ to the next
-Channel\* through transformations (such as a Knative Service which processes
-CloudEvents over HTTP). A _Subscription_ controller resolves the addresses of
-transformations (`subscriber`) and destination storage (`reply`) through the
-_Callable_ and _Addressable_ interface contracts, and writes the resolved
-addresses to the _Channel_ in the `channel` reference. _Subscriptions_ do not
-need to specify both a transformation and a storage destination, but at least
-one must be provided.
+**Addressable** resources expose a resource address (HTTP URL) in their `status`
+object. The URL is used as a destination for delivery of events to the resource;
+the exposed URL must implement the [data plane contract](data-plane.md) for
+receiving events.
```

Comment:
Might be worth pointing out that even though the k8s Service does not meet this, it's been special cased to make it easier to use them natively.

Reply:
Added the words fro this to Destination.